plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java

```diff
@@ -244,8 +246,12 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<Delete
     @VisibleForTesting
     void deleteModelChunks(String modelId, Boolean isHidden, ActionListener<Boolean> actionListener) {
         DeleteByQueryRequest deleteModelsRequest = new DeleteByQueryRequest(ML_MODEL_INDEX);
-        deleteModelsRequest.setQuery(new TermsQueryBuilder(MODEL_ID_FIELD, modelId));
-
+        deleteModelsRequest
+            .setQuery(
+                new BoolQueryBuilder()
+                    .must(new TermsQueryBuilder(MODEL_ID_FIELD, modelId)) // Match documents with the same model_id
+                    .mustNot(new TermQueryBuilder("_id", modelId)) // exclude the document just deleted
```

Comment:
Could we explain more in the comment why do we need to exclude the document just deleted? 
In addition, why is this affecting multi-tenancy use case as this `deleteModelChunks` is for local models.

Reply:
> Could we explain more in the comment why do we need to exclude the document just deleted?

Sure, I'll add a comment.

Search results aren't always up to date until a refresh.  In this case we were deleting an object and then immediately searching.  I think this bug was triggered by an edge case where there are no chunks, but one (deleted) document matched the search.

I don't think it's wrong to make this change; seems a marginal possible performance improvement with one less search hit.

> In addition, why is this affecting multi-tenancy use case as this deleteModelChunks is for local models.

This was fixed before you added code to skip deleteModelChunks in this case.  This fixed that bug before catching up with your change, which is still good for performance reasons.

But it's a correct/general fix that should generally apply.